widget/table.go

```diff
@@ -297,13 +297,21 @@ func (t *Table) Select(id TableCellID) {
 func (t *Table) SetColumnWidth(id int, width float32) {
 	t.propertyLock.Lock()
 	if id < 0 {
+		if t.headerSize.Width == width {
+			t.propertyLock.Unlock()
+			return
+		}
 		t.headerSize.Width = width
 	}
 
 	if t.columnWidths == nil {
 		t.columnWidths = make(map[int]float32)
 	}
 
+	if t.columnWidths[id] == width {
```

Comment:
Should we add an `else if width == templateCell.Width` here? ie if setting column width to the same thing as the default width it is a no-op (same for row height)

Reply:
I don't think that would work, because the `columnWidths[id]` being set would mean it does not use the template wouldn't it?
I guess we could add an optimisation to remove custom widths when they are the same as template, but I'd say that is separate to this fix...

Reply:
I meant the else case where `id` was not an entry in t.columnWidths. Now that I look again, I see that you are blindly accessing t.columnWidths[id] when it may be unset - that may be the cause of the unit test failures.

Reply:
Ah I see, thanks for the catch. I'll get that addressed.

Reply:
> that may be the cause of the unit test failures

The test failures seem to relate to something strange in the release branch merge. I cannot replicate it locally either so I am struggling to find the cause. It's on the main develop branch as well as this PR.